fiware_cloto/cloto/manager/InfoManager.py

```diff
@@ -89,3 +96,46 @@ def init_information(self):
         s = ServerInfo(id=1, owner=settings.OWNER, version=settings.VERSION,
                        runningfrom=runningfrom, doc=settings.API_INFO_URL)
         s.save()
+
+    def publish_message(self, message):
+        """Publish a message related to the windowsize in the rabbitmq and
+        close the connection
+
+        :param str message:       The well-formatted message to send
+        """
+        """ Initialize the class and create a connection with a RabbitMQ server instance.
+        """
+        # Get the default IP address of the RabbitMQ server.
+        rabbitmq = settings.RABBITMQ_URL
+        self.connection = None
+        self.channel = None
+        # Open a remote connection to RabbitMQ on specified IP address
+        try:
+            self.connection = pika.BlockingConnection(pika.ConnectionParameters(
+                host=rabbitmq))
+
+            # Open the channel
+            self.channel = self.connection.channel()
+        except (pika.exceptions.AMQPConnectionError, pika.exceptions.AMQPChannelError), error:
+            raise Exception("Error with Rabbit connection %s", error)
+
+        if self.channel:
+            self.channel.exchange_declare(exchange="windowsizes",
```

Comment:
According to Pika documentation:

      This method creates an exchange if it does not already exist, 
      and if the exchange exists, verifies that it is of the correct and expected class.

      If passive set, the server will reply with Declare-Ok if the exchange already 
      exists with the same name, and raise an error if not and if the exchange 
      does not already exist, the server MUST raise a channel exception with 
      reply code 404 (not found).

As you are declaring the exchange with the same name and class that Facts does, you are right and no exception should be raised.

NTC